core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/DataProcessor.scala

```diff
@@ -203,8 +196,8 @@ class DataProcessor( // dependencies:
   }
 
   def shutdown(): Unit = {
-    dpThread.cancel(true) // interrupt
     operator.close() // close operator
+    dpThread.cancel(true) // interrupt
```

Comment:
does this change fix the UDF apache arrow rpc not shut down error? I remembered that @shengquan-ni wrote this code in the original specific order also for a reason. Please confirm with @shengquan-ni  about this change.

Reply:
Yes. the logic is that `dpThread.cancel(true)` will send `INTERUPTION`, and we believe `ApacheArrow` is handling this signal and thus after this, Arrow client-side is terminated thus not able to send any more instruction to Python side (including `close` and `terminate`, which is invoked by `operator.closed()` here). By swapping them, we first invoke `operator.close()` to let the python side know we are shutting down, then broadcast `INTERUPTION`.  This way python side is terminated peacefully.